grails-app/controllers/au/org/emii/portal/LayerController.groovy

```diff
@@ -117,4 +116,9 @@ class LayerController {
     def _isXmlContent(contentType) {
         return contentType.find(/(text|application)\/xml/)
     }
+
+    def _contentTypeCharset(contentType) {
+        def stringComponents = contentType.split("=")
+        return stringComponents[1]
```

Comment:
And this will return you a bunch of stuff you might not want. So generally speaking you should parse it properly or not parse it at all.

In the case of `Content-Type: application/xml;charset=UTF-8`, this will actually work fine, however think about
`Content-Type: application/xml;charset=UTF-8;value=booga`. Then in your case you gonna return `UTF-8;value` which is undesirable.